cmd/containerd-shim-runhcs-v1/task_jobobject_update.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/Microsoft/hcsshim/internal/jobobject"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/processorinfo"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	jobObjUtilExeName = "jobobj-util.exe"
+
+	cpuLimitFlag     = "cpu-limit"
+	cpuWeightFlag    = "cpu-weight"
+	useNTVariantFlag = "use-nt"
+)
+
+func calculateJobCPURate(ctx context.Context, requestCount uint32) uint32 {
+	hostProcs := uint32(processorinfo.ProcessorCount())
+	if requestCount > hostProcs {
+		log.G(ctx).WithFields(logrus.Fields{
+			"requested": requestCount,
+			"assigned":  hostProcs,
+		}).Warn("Changing user requested CPUCount to current number of processors")
+		requestCount = hostProcs
+	}
+
+	rate := (requestCount * 10000) / hostProcs
+	if rate == 0 {
+		return 1
+	}
+	return rate
+}
+
+func calculateJobCPUWeight(processorWeight uint32) uint32 {
+	return 1 + uint32((8*processorWeight)/jobobject.CPUWeightMax)
+}
```

Comment:
Crud, I don't like having to remake these as they already exist in /internal/jobcontainer but it feels weird to export them also

Reply:
Thoughts?